storage: de-flake TestStoreSplitTimestampCacheDifferentLeaseholder

tbg opened the pull request:
The test prohibited lease requests for the RHS of the split, but in fact
such requests can happen if the test runs for long enough (and the lease
installed by the split expires).

This was exacerbated by #9550, which for this test (but likely in some
generality) means that the RHS of the split does not campaign eagerly
and locks up for the duration of a Raft election trigger.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9721)

<!-- Reviewable:end -->

tamird commented:
Reviewed 3 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[storage/client_split_test.go, line 1188 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5ZKxeEVlz4_2bKUc:-KTG5ZKxeEVlz4_2bKUd:bkiejzn) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/storage/client_split_test.go#L1188)):*
> ```Go
> 		log.Infof(context.TODO(), "received lease request (%s, %s)",
> 			leaseReq.Span, leaseReq.Lease)
> 		debug.PrintStack()
> ```

this is removed in the wrong commit.

---

*[storage/client_split_test.go, line 1252 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5vEy6LuRHg3vDLx9:-KTG5gMJlTqZvronWJrI:b-jk24wz) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9fec20a7790cd8dc43cf4f3b8d1ab695563e7db/storage/client_split_test.go#L1252)):*
> ```Go
> 			t.Fatal(err)
> 		}
> 		leaseHolder, err := tc.FindRangeLeaseHolder(&desc, nil)
> ```

keep the old?

---

*[storage/client_split_test.go, line 1281 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5kAXO03Qp9-Kj79w:-KTG5kAXO03Qp9-Kj79x:b-wyt0w2) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/storage/client_split_test.go#L1281)):*
> ```Go
> 	// In practice, this should only be possible if second-long delays occur
> 	// just above this comment, and we assert against it below.
> 	log.Infof(context.TODO(), "splitting at %s", splitKey)
> ```

TODO or Background? pick one

---

*[storage/store_pool.go, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG6sZVl28b7oCg6cZx:-KTG6sZVl28b7oCg6cZy:b-9unh77) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b67997fba5ce0c6568ae3245b4e0c35fd35b523/storage/store_pool.go#L60)):*
> ```Go
> 	// defaultReserveRPCTimeout is used for the rpc calls to Reserve on other
> 	// nodes. It should be short as this may block calls to ChangeReplicas.
> 	defaultReserveRPCTimeout = 8 * time.Second
> ```

@bramgruneir should look at this, probably.

---

*[testutils/testcluster/testcluster.go, line 412 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG61gldrpLtYpSxejp:-KTG61gldrpLtYpSxejq:bb1jz7c) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/testutils/testcluster/testcluster.go#L412)):*
> ```Go
> func (tc *TestCluster) FindRangeLease(
> 	rangeDesc *roachpb.RangeDescriptor, hint *ReplicationTarget,
> ) (_ *roachpb.Lease, now hlc.Timestamp, _ error) {
> ```

suggest removing named returns

actually, a bunch of these changes seem unrelated to the actual fix here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: de-flake TestStoreSplitTimestampCacheDifferentLeaseholder

The test prohibited lease requests for the RHS of the split, but in fact
such requests can happen if the test runs for long enough (and the lease
installed by the split expires).

This was exacerbated by #9550, which for this test (but likely in some
generality) means that the RHS of the split does not campaign eagerly
and locks up for the duration of a Raft election trigger.

tschottdorf commented:
Review status: 1 of 3 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[storage/client_split_test.go, line 1188 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5ZKxeEVlz4_2bKUc:-KTGAgWvfZkhhGLv5r_P:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/storage/client_split_test.go#L1188)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is removed in the wrong commit.
</details>
Done.

---

*[storage/client_split_test.go, line 1252 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5vEy6LuRHg3vDLx9:-KTGAibUlSQIgKruB84Z:bmq68zu) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9fec20a7790cd8dc43cf4f3b8d1ab695563e7db/storage/client_split_test.go#L1252)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> keep the old?
</details>
This line is the reason for introducing the new method as `FindRangeLeaseHolder` returns an error when the lease isn't active any more.

---

*[storage/client_split_test.go, line 1281 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG5kAXO03Qp9-Kj79w:-KTGBwKRosHneATkBvWd:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/storage/client_split_test.go#L1281)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> TODO or Background? pick one
</details>
Done.

---

*[storage/store_pool.go, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG6sZVl28b7oCg6cZx:-KTGCMxeTfE-H5WAldBy:b-pbdqz3) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b67997fba5ce0c6568ae3245b4e0c35fd35b523/storage/store_pool.go#L60)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> @bramgruneir should look at this, probably.
</details>
Agreed.

---

*[testutils/testcluster/testcluster.go, line 412 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG61gldrpLtYpSxejp:-KTGC1cMdAXLcwMmJd9U:b6hqts) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d17df3ca28487397af6561a14a580247c00130/testutils/testcluster/testcluster.go#L412)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> suggest removing named returns
> 
> actually, a bunch of these changes seem unrelated to the actual fix here.
</details>
No, introducing the new method is part of the fix as the old one would error out when the lease was expired.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> mod the 1s -> 8s.

---

Reviewed 2 of 2 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-:-KTGEKjPtLEMUtgV5DZB:bzf005i)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Reviewed 1 of 2 files at r4.
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[storage/store_pool.go, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG6sZVl28b7oCg6cZx:-KTGIyi-tPy1DcIy8iTC:b-n47ex4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b67997fba5ce0c6568ae3245b4e0c35fd35b523/storage/store_pool.go#L60)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Agreed.
</details>
How did you choose 8s instead of say 10s?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-:-KTGJGyzd86qd0Sln4PK:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[storage/store_pool.go, line 60 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721#-KTG6sZVl28b7oCg6cZx:-KTGLnNFQYO7my3vpWMH:b3fi7gx) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b67997fba5ce0c6568ae3245b4e0c35fd35b523/storage/store_pool.go#L60)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> How did you choose 8s instead of say 10s?
</details>
By the same method you probably used for the `1s`: I just picked a number. I'm wondering whether this timeout is useful at all, but don't want to pull on that thread too much at this point.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9721)*
<!-- Sent from Reviewable.io -->
